core/src/toga/__init__.py

```diff
@@ -1,47 +1,71 @@
+    "ActivityIndicator": "toga.widgets.activityindicator",
+    "Widget": "toga.widgets.base",
+    "Box": "toga.widgets.box",
+    "Button": "toga.widgets.button",
+    "Canvas": "toga.widgets.canvas",
+    "DateInput": "toga.widgets.dateinput",
+    "DatePicker": "toga.widgets.dateinput",
+    "DetailedList": "toga.widgets.detailedlist",
+    "Divider": "toga.widgets.divider",
+    "ImageView": "toga.widgets.imageview",
+    "Label": "toga.widgets.label",
+    "MapPin": "toga.widgets.mapview",
+    "MapView": "toga.widgets.mapview",
+    "MultilineTextInput": "toga.widgets.multilinetextinput",
+    "NumberInput": "toga.widgets.numberinput",
+    "OptionContainer": "toga.widgets.optioncontainer",
+    "OptionItem": "toga.widgets.optioncontainer",
+    "PasswordInput": "toga.widgets.passwordinput",
+    "ProgressBar": "toga.widgets.progressbar",
+    "ScrollContainer": "toga.widgets.scrollcontainer",
+    "Selection": "toga.widgets.selection",
+    "Slider": "toga.widgets.slider",
+    "SplitContainer": "toga.widgets.splitcontainer",
+    "Switch": "toga.widgets.switch",
+    "Table": "toga.widgets.table",
+    "TextInput": "toga.widgets.textinput",
+    "TimeInput": "toga.widgets.timeinput",
+    "TimePicker": "toga.widgets.timeinput",
+    "Tree": "toga.widgets.tree",
+    "WebView": "toga.widgets.webview",
+    "Window": "toga.window",
+}
+for name in toga_core_imports:
+    __all__.append(name)
 
-# Types
-from .types import LatLng
 
-# Widgets
-from .widgets.activityindicator import ActivityIndicator
-from .widgets.base import Widget
-from .widgets.box import Box
-from .widgets.button import Button
-from .widgets.canvas import Canvas
-from .widgets.dateinput import DateInput, DatePicker
-from .widgets.detailedlist import DetailedList
-from .widgets.divider import Divider
-from .widgets.imageview import ImageView
-from .widgets.label import Label
-from .widgets.mapview import MapPin, MapView
-from .widgets.multilinetextinput import MultilineTextInput
-from .widgets.numberinput import NumberInput
-from .widgets.optioncontainer import OptionContainer, OptionItem
-from .widgets.passwordinput import PasswordInput
-from .widgets.progressbar import ProgressBar
-from .widgets.scrollcontainer import ScrollContainer
-from .widgets.selection import Selection
-from .widgets.slider import Slider
-from .widgets.splitcontainer import SplitContainer
-from .widgets.switch import Switch
-from .widgets.table import Table
-from .widgets.textinput import TextInput
-from .widgets.timeinput import TimeInput, TimePicker
-from .widgets.tree import Tree
-from .widgets.webview import WebView
-from .window import Window
+def __getattr__(name):
+    if name in toga_core_imports:
+        module = importlib.import_module(f"{toga_core_imports[name]}")
+        globals()[name] = getattr(module, name)
+        # __all__.append(name)
+        return getattr(module, name)
+    else:
+        raise AttributeError(f"module {__name__} has no attribute {name}")
```

Comment:
To match the standard exception message, there should be quotes around the names:
```
>>> import json
>>> json.foo
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'json' has no attribute 'foo'
```